source/common/protobuf/utility.h

```diff
@@ -84,6 +84,15 @@ uint64_t fractionalPercentDenominatorToInt(
 } // namespace ProtobufPercentHelper
 } // namespace Envoy
 
+// Convert an envoy::api::v2::core::Percent to a double or a default.
+// @param message supplies the proto message containing the field.
+// @param field_name supplies the field name in the message.
+// @param default_value supplies the default if the field is not present.
+#define PROTOBUF_PERCENT_TO_DOUBLE_OR_DEFAULT(message, field_name, default_value)                  \
+  (!std::isnan((message).field_name().value())                                                     \
+       ? (message).has_##field_name() ? (message).field_name().value() : default_value             \
+       : throw EnvoyException(fmt::format("Value not in the range of 0..100 range.")))
```

Comment:
Please add an explicit test for this error and general usage in a follow up somewhere in the protobuf utility tests.